File: backend/app/models/budget.py
from datetime import datetime

from sqlalchemy import DateTime, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Budget(Base):
    __tablename__ = "budgets"

    id: Mapped[int] = mapped_column(primary_key=True)
    category: Mapped[str] = mapped_column(String(50), unique=True, index=True)
    amount: Mapped[int]
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now()
    )

File: frontend/src/types/budget.ts
/**
 * バックエンドの BudgetUpsert スキーマ（backend/app/schemas/budget.py）に対応するリクエスト型。
 */
export interface BudgetUpsertPayload {
  amount: number;
}

/**
 * バックエンドの BudgetRead スキーマに対応するレスポンス型。
 */
export interface BudgetApiResponse {
  id: number;
  category: string;
  amount: number;
  created_at: string;
  updated_at: string;
}
